src/Core/Context.cpp

```diff
@@ -102,7 +102,13 @@ void Context::Update() {
     Util::Input::Update();
     SDL_GL_SwapWindow(m_Window);
     glClear(GL_COLOR_BUFFER_BIT | GL_DEPTH_BUFFER_BIT);
+
+    if constexpr (FPS_CAP != 0) {
+        constexpr double frameTime = 1000 / static_cast<double>(FPS_CAP);
+        SDL_Delay(static_cast<Uint32>(frameTime - Util::Time::GetDeltaTime()));
+    }
 }
```

Comment:
The `Util::Time::GetDeltaTime()` is getting the delta time between "last frame start time" and "current frame start time", Which is not the delta time between "current frame start" and the moment "this function got called". Which cause it always Delay for wrong period.